AK/Utf8View.cpp

```diff
@@ -143,6 +153,24 @@ Utf8View Utf8View::trim(Utf8View const& characters, TrimMode mode) const
     return substring_view(substring_start, substring_length);
 }
 
+bool Utf8View::validate(size_t& valid_bytes, AllowSurrogates allow_surrogates) const
+{
+    auto result = simdutf::validate_utf8_with_errors(m_string.characters_without_null_termination(), m_string.length());
+    valid_bytes = result.count;
+
+    if (result.error == simdutf::SURROGATE && allow_surrogates == AllowSurrogates::Yes) {
+        valid_bytes += 3; // All surrogates have a UTF-8 byte length of 3.
+
+        size_t substring_valid_bytes = 0;
+        auto result = substring_view(valid_bytes).validate(substring_valid_bytes, allow_surrogates);
```

Comment:
feels awkward to shadow `result` here, given how below you return `result.error == some value` and on 168 you return `result` directly, but they have two different types.

Reply:
woops - I'm surprised I didn't get an error about this, I thought we had -Wshadow (or whatever the warning is) on